woocommerce/woocommerce
# Marketplace: My Subscriptions feature branch

This is the feature branch for the rebuild of the `WooCommerce > Extensions > My Subscription` section. @woocommerce/alpha-fire are responsible for it.

### Feature flag

The changes in this PR are behind the same feature flag that enables the React-powered marketplace: `WooCommerce > Settings > Advanced > New, faster way to find extensions and themes for your WooCommerce store`. This is enabled by default.

### Changes proposed in this Pull Request:

This PR contains these changes, which comprise the new version of My Subscriptions:

|PR|Description|
|---|---|
|https://github.com/woocommerce/woocommerce/pull/40341|WooCommerce > Extensions > My subscriptions skeleton|
|https://github.com/woocommerce/woocommerce/pull/40433|Add marketplace helper subscriptions API|
|https://github.com/woocommerce/woocommerce/pull/40499|Add Marketplace license management endpoints|
|https://github.com/woocommerce/woocommerce/pull/40626|Marketplace: Implement the subscriptions table row|
|https://github.com/woocommerce/woocommerce/pull/40727|Return to new subscriptions page after action|
|https://github.com/woocommerce/woocommerce/pull/40703|Add Marketplace subscriptions API license error responses|
|https://github.com/woocommerce/woocommerce/pull/40820|Support free products in My Subscriptions|
|https://github.com/woocommerce/woocommerce/pull/40862|Add subscriptions empty state|
|https://github.com/woocommerce/woocommerce/pull/40752|My subscription product updates|
|https://github.com/woocommerce/woocommerce/pull/41127|Change woocommerce.com URLs to woo.com in marketplace|
|https://github.com/woocommerce/woocommerce/pull/40990|Update/my subscriptions wp installer|
|https://github.com/woocommerce/woocommerce/pull/41124|My subscriptions error notices|
|https://github.com/woocommerce/woocommerce/pull/41066|Marketplace: table design updates|
|https://github.com/woocommerce/woocommerce/pull/41288|Fix subscription grouping|
|https://github.com/woocommerce/woocommerce/pull/41267|Marketplace - Update empty state for installed subscriptions table|
|https://github.com/woocommerce/woocommerce/pull/41397|Update refresh button to use API|
|https://github.com/woocommerce/woocommerce/pull/41398|Marketplace connection deny redirect|
|https://github.com/woocommerce/woocommerce/pull/41395|Check if wc_admin_register_page exists before loading Marketplace pages|
|https://github.com/woocommerce/woocommerce/pull/41455|Marketplace: Add Tracks events to My Subscriptions page|
|https://github.com/woocommerce/woocommerce/pull/41365|Marketplace - tracking enhancements|
|https://github.com/woocommerce/woocommerce/pull/41630|Marketplace: fix sideways scrolling of the my subscriptions table on Chromium based browsers|
|https://github.com/woocommerce/woocommerce/pull/41634|My Subscriptions: link product icon and product name to Woo.com product page|

The PR also includes these changes, which are updates to the Marketplace generally.

|PR|Description|
|---|---|
|https://github.com/woocommerce/woocommerce/pull/41633|Add Woo button to plugins list|
|https://github.com/woocommerce/woocommerce/pull/41603|Marketplace: update no results page| 
|https://github.com/woocommerce/woocommerce/pull/41600|Add Marketplace readme|



### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Please check out this branch and run a build.
2. View the [My Subscriptions](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions) screen, or browse to `WooCommerce > Extensions > My Subscriptions`.
3. Connect your store to Woo.com.
4. Open the Woo.com [My Subscriptions](https://woo.com/my-account/my-subscriptions/) page in a separate browser tab, and compare with the list of subscriptions you see.  There will be some slight differences - subscriptions that expired more than 6 months ago won't appear – but overall, the list in your store should correspond to the list on Woo.com.
5. Test the various actions you can take with subscriptions. (Actions are available on certain subscriptions based on their state.)
* Connect the subscription (that is, activate the license on your store).
* View documentation.
* Manage on Woo.com.
* Manage in Plugins.
* Subscribe (that is, buy a license) for an extension you've installed on the store, but don't have a subscription for.
* Install the extension.
6. Try updating your subscription list by clicking the "Refresh" button in top right.
7. Disconnect your store using the `Disconnect account` option from the user menu in top right of the page, then test that you can reconnect.

For more detailed instructions, please see the individual PRs.

### Changelog entry



-   [] Automatically create a changelog entry from the details below.

<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
